sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEvent.java

```diff
@@ -0,0 +1,470 @@
+// Licensed under the MIT License.
+
+package com.azure.core.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.JsonSerializer;
+import com.azure.core.util.serializer.JsonSerializerProviders;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.core.util.serializer.TypeReference;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * The CloudEvent model. This represents a cloud event as specified by the
+ * <a href="https://github.com/cloudevents/spec/blob/v1.0.1/spec.md">Cloud Native Computing Foundation</a>
+ */
+@Fluent
+public final class CloudEvent {
+    private static final String SPEC_VERSION = "1.0";
+    private static final JsonSerializer SERIALIZER;
+
+    static {
+        JsonSerializer tmp;
+        try {
+            tmp = JsonSerializerProviders.createInstance();
+        } catch (IllegalStateException e) {
+            tmp = new JacksonSerializer();
+        }
+        SERIALIZER = tmp;
+    }
+
+    private static final ClientLogger LOGGER = new ClientLogger(CloudEvent.class);
+    private static final Set<String> RESERVED_ATTRIBUTE_NAMES = new HashSet<>(Arrays.asList(
+        "specversion",
+        "id",
+        "source",
+        "type",
+        "datacontenttype",
+        "dataschema",
+        "subject",
+        "time",
+        "data"
+    ));
+
+    /*
+     * An identifier for the event. The combination of id and source must be
+     * unique for each distinct event.
+     */
+    @JsonProperty(value = "id", required = true)
+    private String id;
+
+    /*
+     * Identifies the context in which an event happened. The combination of id
+     * and source must be unique for each distinct event.
+     */
+    @JsonProperty(value = "source", required = true)
+    private String source;
+
+    /*
+     * Event data specific to the event type.
+     */
+    @JsonProperty(value = "data")
+    private Object data;
+
+    /*
+     * Event data specific to the event type, encoded as a base64 string.
+     */
+    @JsonProperty(value = "data_base64")
+    private String dataBase64;
```

Comment:
Is this required to expose this as a separate `String` variable ? You already have `data`  variable. I see that all you are doing is base64 encoding, you can store  encoded value in `data` variable.  I do not know if this is needed by backend ?

Reply:
Yes, the CloudEvent spec use "data" for json/string data and "data_base64" for bytes. We must separate it in the model class.
But we expose only `BinaryData getData()` to users.